### PR TITLE
Add protocol impl

### PR DESCRIPTION
**Pull request by wcheng-msft**
Add support for protocol extensions

### CODE REVIEW

File: preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp [hunk 1]
```diff
@@ -52,13 +105,91 @@ HRESULT Protocol::ExecuteForAddRequest()
...
+        RETURN_IF_FAILED(ProcessProtocolForRemove(*protocol));
+    }
+
+    return S_OK;
+}
+
+HRESULT Protocol::ProcessProtocolForRemove(ProtocolData& protocol)
+{
+    HRESULT hrDeleteKey = m_classesKey.DeleteTree(protocol.name.c_str());
+    if (FAILED(hrDeleteKey))
```
Comment (rudyhuyn) [inline preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp]: Probably check if the protocol is associated to the application we try to uninstall (verify the value of `open/command` first before removing the key)

File: preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp [hunk 2]
```diff
@@ -52,13 +105,91 @@ HRESULT Protocol::ExecuteForAddRequest()
     return S_OK;
 }
 
+HRESULT Protocol::ExecuteForAddRequest()
+{
+    for (std::vector<ProtocolData>::iterator protocol = m_protocols.begin(); protocol != m_protocols.end(); ++protocol)
```
Comment (msftrubengu) [inline preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp]: auto?

File: preview/Win7Msix/Win7MSIXInstaller/Protocol.hpp [hunk 3]
```diff
@@ -2,20 +2,41 @@
...
+#include "RegistryKey.hpp"
+#include <vector>
+
+/// Data structs to be filled in from the information in the manifest
+struct ProtocolData
+{
+    std::wstring name;
+    std::wstring parameters;
+    std::wstring displayName;
+    std::wstring logo;
```
Comment (sarjanas-msft) [inline preview/Win7Msix/Win7MSIXInstaller/Protocol.hpp]: Recommend adding /* inline */ comments to explain each member of the struct.

**Comment by sarjanas-msft** [comment]
Add the test to the PR

File: preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp (hunk 2 above)
Comment (wcheng-msft) [reply preview/Win7Msix/Win7MSIXInstaller/Protocol.cpp]: fixed here and for FTA as well

---
In reply to: [265693744](https://github.com/Microsoft/msix-packaging/pull/67#discussion_r265693744) [](ancestors = 265693744)

**Pull request merged**